pipeline/src/lib.rs

```diff
@@ -108,15 +170,20 @@ pub fn handle_simple_queries_callback<'a, T: FieldElement>() -> impl QueryCallba
         let (id, data) = parse_query(query)?;
         match id {
             "None" => Ok(None),
-            "PrintChar" => {
-                assert_eq!(data.len(), 1);
-                print!(
-                    "{}",
-                    data[0]
-                        .parse::<u8>()
-                        .map_err(|e| format!("Invalid char to print: {e}"))?
-                        as char
-                );
+            "Output" => {
+                assert_eq!(data.len(), 2);
+                let fd = data[0]
+                    .parse::<u32>()
+                    .map_err(|e| format!("Invalid fd: {e}"))?;
+                let byte = data[1]
+                    .parse::<u8>()
```

Comment:
I'm a bit confused, the syscall writes `u32`, how come here it is parse `u8` for the data?

Reply:
this data here is only to print chars, so this must be a char

Reply:
the point is that the main ways we use this are to either print it to stdout, or write into a Deserializable struct, so in practice we always use u8. We can change it to make it more granular, but I'd do it in a follow-up PR

Reply:
I mean, why isn't the method simply called `write_u8`?

Reply:
yea good point. I'll make the distinction more precise

Reply:
pushed